### PR TITLE
chore: tests for columns-modify-dialog component and some refactor

### PR DESCRIPTION
**Pull request by Faust1-2was-Aneo**

### CODE REVIEW

File: src/app/components/columns-modify-dialog.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      component.updateColumn({checked: true} as MatCheckboxChange, 'id' as ColumnKey<object, object>);
+      expect(component.columns).toEqual(['id', 'created_time']);
+    });
+
+    it('should not push a column already that is not in the available columns', () => {
+      component.updateColumn({checked: true} as MatCheckboxChange, 'someColumn' as ColumnKey<object, object>);
+      expect(component.columns).toEqual(['id', 'created_time']);
+    });
+
+    it('should not do anything if a column not in the column list is removed', () => {
```
Comment (yannick-aneo) [inline src/app/components/columns-modify-dialog.component.spec.ts:93]: I didn't understand. 
Maybe you meant "should not do anything if a column is not in the column list" ?

Comment (Faust1-2was-Aneo) [reply src/app/components/columns-modify-dialog.component.spec.ts:93]: i will try to clarifiate it.

File: src/app/components/columns-modify-dialog.component.ts [hunk 2]
```diff
@@ -99,12 +99,20 @@ export class ColumnsModifyDialogComponent<T extends object,O extends object> imp
...
    * Update the columns array when a checkbox is checked or unchecked
+   * checked: add the column.
+   * Unchecked: remove it
    */
   updateColumn({ checked }: MatCheckboxChange, column: ColumnKey<T, O>): void {
     if (checked) {
-      this.columns.push(column);
+      if (!this.columns.includes(column) && this.data.availableColumns.includes(column)) {
+        this.columns.push(column);
+        this.data.availableColumns = this.data.availableColumns.filter(c => c !== column);
```
Comment (yannick-aneo) [inline src/app/components/columns-modify-dialog.component.ts:109]: I guess it's better to give a more meaningful name to variable c in Filter method.

File: src/app/components/columns-modify-dialog.component.ts [hunk 3]
```diff
@@ -99,12 +99,20 @@ export class ColumnsModifyDialogComponent<T extends object,O extends object> imp
...
+      }
     } else {
-      this.columns = this.columns.filter(c => c !== column);
+      if(this.columns.includes(column)) {
```
Comment (yannick-aneo) [inline src/app/components/columns-modify-dialog.component.ts:112]: Same here

**Review by yannick-aneo — CHANGES_REQUESTED** [review]

**Review by GunaDinesh — APPROVED** [review]

**Pull request merged**